spmd/tensor/ops/common_rules.py

```diff
@@ -147,6 +148,29 @@ def merge_sharding(dim: str, a: int, b: int) -> int:
             if value != len(input_specs):
                 needs_reshard = True
 
+    for mesh_dim, dims in seen_shardings.items():
```

Comment:
Do we align broadcast dimensions before doing this?

Reply:
E.g. when we are summing 

A(20,6)[shard(1)]  + B(6)[shard[0])

the shards are aligned . Just makeing sure that this is taken into account.

Reply:
It should be handled in pointwise ops before calling the einop rule, logic [here](https://github.com/pytorch/tau/blob/main/spmd/tensor/ops/common_rules.py#L180), also added this as a test case in https://github.com/pytorch/tau/pull/496/files#diff-d12eecf13eb5e75f53a3542af9b905a09717da64a3586df228c0502a61c17492R279